Classes/Configuration/PhinxConfiguration.php

```diff
@@ -28,22 +32,49 @@ public function toArray(): array
             'environments' => [
                 'default_migration_table' => self::MIGRATION_TABLE_NAME,
                 'default_environment' => 'typo3',
-                'typo3' => [
-                    'adapter' => 'mysql',
-                    'host' => $connectionParameters['host'],
-                    'port' => $connectionParameters['port'],
-                    'user' => $connectionParameters['user'],
-                    'pass' => $connectionParameters['password'],
-                    'name' => $connectionParameters['dbname'],
-                    'unix_socket' => $connectionParameters['unix_socket'] ?? null,
-                    'charset' => $connectionParameters['charset'],
-                ],
+                'typo3' => $this->getEnvironment($connection),
             ],
         ];
     }
 
-    private function getConnectionParameters(): array
+    private function getEnvironment(Connection $connection): array
     {
-        return $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default'];
+        switch ($connection->getDatabasePlatform()->getName()) {
+            case 'sqlite':
+                $pathInfo = pathinfo($connection->getDatabase());
+                $name = $pathInfo['dirname'] . '/' . $pathInfo['filename'];
+                $suffix = '.' . $pathInfo['extension'];
+
+                return [
+                    'adapter' => 'sqlite',
+                    'name' => $name,
+                    'suffix' => $suffix,
+                ];
+            case 'mssql':
+                $adapter = 'sqlsrv';
+                break;
+
+            case 'postgresql':
+                $adapter = 'pgsql';
+                break;
+
+            case 'mysql':
+                $adapter = 'mysql';
+                break;
+
+            default:
+                return [];
+        }
+
+        return [
+            'adapter' => $adapter,
+            'host' => $connection->getParams()['host'] ?? null,
```

Comment:
I'd at least call `getParams()` only once. Also it seems like we can basically return that `$params` then.

Finally `getParams()` is marked as `@internal` so we should not rely on it.